kernel/typeops.ml

```diff
@@ -432,21 +432,8 @@ and execute_array env = Array.map (execute env)
 
 (* Derived functions *)
 
-let universe_levels_of_constr _env c =
-  let rec aux s c =
-    match kind c with
-    | Const (_c, u) ->
-       LSet.fold LSet.add (Instance.levels u) s
-    | Ind ((_mind,_), u) | Construct (((_mind,_),_), u) ->
-       LSet.fold LSet.add (Instance.levels u) s
-    | Sort u when not (Sorts.is_small u) ->
-      let u = Sorts.univ_of_sort u in
-      LSet.fold LSet.add (Universe.levels u) s
-    | _ -> Constr.fold aux s c
-  in aux LSet.empty c
```

Comment:
universes_of_constr was already in the kernel, see this part of the diff

Reply:
Bah, I missed that. I guess that's OK then.

Reply:
It is used to check the universes are declared mainly, what's wrong with that?

Reply:
I read it too quickly and thought that was the old version that was randomly picking universes declared in the constants. The current version is sane indeed.